### ISSUE TITLE
Dagger 0.3.0-alpha.1 release

### ISSUE THREAD

**Issue by mircubed** [dagger/dagger#3283]
Dagger `0.3.0-alpha.1` Release
Target date: **Oct 19, 2022**
Dagger `0.3.0-alpha.1` is a planned release featuring the bare minimum to allow the community to try the new cloak engine and have a good experience.

### Success Criteria:

- [x] User can successfully install dagger `0.3.0-alpha.1` starting from the GitHub changelog or release notes (no announcement coming right at release date) - disregard #3282 
- [ ] User can easily switch back and forth between dagger 0.2 and 0.3.0-alpha.1 - #3176
- [ ] If a user gets stuck while using Dagger `0.3.0-alpha.1` - #3316 

#### Docs
- [ ] Docs versioning plan - #3318 
- [ ] Docs launch plan (can be done between release and launch) #3440 

#### Go SDK
- [ ] User can write a dagger script in Go, and enjoy the experience - 
- [x] This PR needs to be merged: #3125
- [x] We need to decide on whether or not we are going to include the core API with the dagger repo or have users codegen it from the start. Then based on the decision we need to do followup work. Made a placeholder issue for that here #3320

### Task list

This was copied from #3176 so that we consolidate these two issues. We want to keep everything in this issue, as discussed with @shykes.

This is the list of tasks required to complete the release of Dagger 0.3.0-alpha.1:
- [x] #3261
- [x] #3285
- [x] #3310
- [ ] Decide if this release includes SDKs, or only the engine #3271 
- [ ] #3347 Define CLI design - STUCK, DON'T MERGE RELEASE UNTIL DONE 
- [x] GitHub Actions workflow that publishes `dagger v0.3.0-alpha.N` pre-release #3356
- [x] Ensure that `install.sh` script works on Linux with `v0.3.0-alpha.N` versions
- [x] Check that `main` docs were imported into `cloak` #3371
- [x] Check that any automation from `main` that we need in `cloak` was imported #3374
- [x] Merge `cloak` into `main` - @aluzzardi took care of that
- [x] Create a new `v0.2.x` branch from the commit just before the merge #3376

As we make progress through these, we will be replacing placeholder text with PRs. There are a few examples above.

The above checklist is not definitive, but it's a good enough snapshot of various discussions and decisions. As we go through the items, we expect to discover new ones. It is also possible that some of the existing ones will become redundant. We will update the list as we make progress.

Last updated on **Oct. 19, 2022**

**Comment by gerhard** [comment]
This reads very well, thank you @mircubed for writing it all down!

> @mircubed Docs: @gerhard @vikram-dagger for the first three bullets here, I don't see them covered in https://github.com/dagger/dagger/issues/3176. Should I make a new issue to ensure this experience is covered?

I've been thinking about this for several days now and I think that we should have a separate issue that tracks the docs changes. Would you like to kick that off @vikram-dagger?

**Comment by gerhard** [comment]
I have made one change which is worth expanding on: `beta` instead of `alpha`. I updated this in the comment above, as well as in the milestone.

I see use producing at least a few `alpha` releases as we test the new release process. As much as I want to think that we will get it right with the first release, experience tells me that this is unlikely to be the case. I am giving us the opportunity to mess up in a safe way with `alpha` releases before we produce the first `beta` release for end-users. I have captured some of my thoughts on `Alphas` vs `Betas` vs `RCs` in this comment: https://github.com/dagger/dagger/pull/3261#issuecomment-1268993644

The approach described in the previous link is something that I would like us to try out as we make progress towards the final `v0.3.0` public release. This is the rough & loose timeline that I imagine (subject to change, of course):

- Oct. 7 - 13 produce at least a few `Alphas` and mature the new release process
- Oct. 14 produce first `Beta` that can be shared with end-users
- Oct. 17 - 31 produce as many `Betas` as we want/need
- Nov. 1 **maybe** produce first `RC`
- Nov. 2-30 produce as many `Betas` as we want/need (RC follow-ups)
- Dec. 1 **definitely** produce first `RC` (maybe the second one)

There is too much uncertainty to talk about the rest of this timeline. What matters is that once we have a few RCs out and everyone is happy with them (especially the users), we could (in theory) product the final `0.3.0` release whenever we feel ready.

What are your thoughts? cc @aluzzardi @samalba @shykes

**Comment by shykes** [comment]
@gerhard we need to call this release `alpha` to avoid interference with the "tech preview" launch story. What we want to tell our community with this launch is "now in tech preview: this cool thing still in development, if you don't mind unfinished code, come join the fun!". If our release said "beta" it would contradict that story, and instead set the expectation that we are launching a beta product (which we are not). In this case it's better to under-promise and over-deliver.

However, you make a good point that we may need to quickly release an `alpha.2`, `alpha.3` etc, to fix unexpected bugs. That is totally fine: thanks to launch/release decoupling, it doesn't matter what the latest available release is at time of launch: as long as it doesn't have the word "beta" in it :)

cc @aluzzardi @samalba

**Comment by gerhard** [comment]
> @gerhard we need to call this release `alpha` to avoid interference with the "tech preview" launch story.

OK, I understand, `v0.3.0-alpha.N` it is.

> @gerhard What we want to tell our community with this launch is "now in tech preview: this cool thing still in development, if you don't mind unfinished code, come join the fun!". If our release said "beta" it would contradict that story, and instead set the expectation that we are launching a beta product (which we are not). In this case it's better to under-promise and over-deliver.

I'm with you.

> However, you make a good point that we may need to quickly release an `alpha.2`, `alpha.3` etc, to fix unexpected bugs. That is totally fine: thanks to launch/release decoupling, it doesn't matter what the latest available release is at time of launch: as long as it doesn't have the word "beta" in it :)

`FIN+ACK`

**Comment by gerhard** [comment]
> just not sure what to call this MVP issue then. @gerhard

`v0.3.0-alpha.N` sounds good to me.

**Comment by mircubed** [comment]
@jpadams @shykes What is the agreement for how users will find our extensions for the engine alpha release since Universe won't be ready simultaneously?

Will it just be the readme for now? Or do we need to do something more before the engine release? Once confirmed, I will add it as a bullet in this MVP.

**Comment by mircubed** [comment]
@gerhard We talked about the release today, so I updated the description based on that. I changed the alpha to alpha.1 since we reduced the scope

**Comment by sipsma** [comment]
> User can write a dagger script in Go, and enjoy the experience - @sipsma @aluzzardi how should we track if this specific experience is being completed?

* This PR needs to be merged: https://github.com/dagger/dagger/pull/3125
* We need to decide on whether or not we are going to include the core API with the dagger repo or have users codegen it from the start. Then based on the decision we need to do followup work. Made a placeholder issue for that here https://github.com/dagger/dagger/issues/3320

**Comment by mircubed** [comment]
@sipsma @aluzzardi @gerhard 

>If a user gets stuck while using Dagger 0.3.0-alpha.1 - because of an error or not knowing what to do - they should be given clear instructions on how to troubleshoot themselves, and how to ask for help - @sipsma @aluzzardi do you already have an issue tracking this work? I couldn't find it

Do we already have a place where we are tracking and working on error message experience for the release or should I create an issue for that? I think we at least need to point the user to where they can get help at the least.

**Comment by sipsma** [comment]
@mircubed I made a placeholder issue yesterday: https://github.com/dagger/dagger/issues/3316

I can work on this quickly. I will need to know where we should actually link users to but I'll write out the code with a placeholder in the meantime.

**Comment by mircubed** [comment]
@gerhard just wanted to make sure that you saw the newest docs plan for launch in #3360. I updated the description in this issue too. The docs content creation might still be happening from the release date to the launch date, but that is ok since we are currently tracking to get it all done by Thursday.

**Comment by mircubed** [comment]
@gerhard adding note here as well - it is essential that we **don't** release this until #3347 is resolved. I know that it is already on the checklist, but just making it clear that is it an important one that we can't skip before the release.

**Comment by gerhard** [comment]
> @gerhard it is essential that we **don't** release this until #3347 is resolved

I understand that we cannot make the release widely public until #3347 is solved.

The agreement was to continue with the release in parallel → https://github.com/dagger/dagger/issues/3347#issuecomment-1277145111

**Comment by gerhard** [comment]
After a sync with @shykes & @aluzzardi, it was decided to pull down the `0.3.0-alpha.1` release published a few days ago. This includes the GitHub tag, release with all the artefacts, Homebrew formula, S3 files, etc.

The artefacts that we will release this week are very different to what we released on Friday. To avoid confusion, we are pulling down the initial release.

cc @vikram-dagger since it will affect the content in this in-flight PR #3406.
